File: Sorting/Bubble_Sort/Bubble_Sort.c
#include <stdio.h>

// Function for bubble sort
void Bubble_Sort(int array[], int size)
{
    int temp, flag;
    for (int i = 0; i < size - 1; i++)
    {
        flag = 0;
        for (int j = 0; j < size - i - 1; j++)
        {
            // Do swapping
            if (array[j] > array[j + 1])
            {
                temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
                flag++;
            }
        }
        if (!flag)
            break;
    }

}

// Function to print elements of array
void Print_Array(int array[], int size)
{
    for (int i = 0; i < size; i++)
        printf("%d ", array[i]);

    printf("\n");
}

// Driver Function
int main()
{
    int array[] = {2, 4, 3, 1, 6, 8, 4};
    Bubble_Sort(array, 7);
    Print_Array(array, 7);
    return 0;
}

// Output
// 1 2 3 4 4 6 8

File: bubblesort.js
function bubble_sort(a)
{
var swapp;
var n=a.length-1;
var x=a;
do
{
swapp=false;
for(var i=0;i<n;i++)
{
if(x[i]<x[i+1])
{
var temp =x[i];
x[i]=x[i+1];
x[i+1]=temp;
swapp =true;
}
}
n--;
}while(swapp);
return x;
}
console.log(bubble_sort([12,34,89,567,986)]));